### DISCUSSION TITLE
Securesocket

### DISCUSSION THREAD

**Discussion by louisvangeldrop** [Moddable-OpenSource/moddable#900] [General]
"https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/network/securesocket.md#memory" mentions the memory-constraints of using TLS. In my program I am reading information from the "aylanetworks.com" server  in chunks of 16KB data resulting in an "memory full" error. Will it help if I buy an ESP32 WRover with 8MB PSRAM?

**Comment by wilberforce** [comment]
Are you talking about the TLS negotation failing as the the chunk size on the the server is too big - or the data you are requesting once the connection has been established?

**Reply by wilberforce** [reply]
You can try this...

https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/network/securesocket.md

The `tls_max_fragment_length` option - try setting to 2048 .

As aylanetworks.com is for iot - they should honor the max fragement size request

**Reply by louisvangeldrop** [reply]
Thanks for the suggestion, but unfortunately it didn't help.  See screenshot for the details:
![image](https://user-images.githubusercontent.com/10716245/163481880-abc26cad-fcda-47c0-acc8-c678c45c63be.png)

**Comment by phoddie** [comment]
Most of the memory used for TLS is allocated in the JavaScript heap. The error message in your debug log indicates that the JavaScript heap is full.

The default allocation for JavaScript heaps on ESP32 is 32 KB. There is plenty of memory on ESP32 to increase that. A practical limit is somewhere between 96 and 128 KB (because of the non-contiguous memory of ESP32), but more than 64 KB is seldom necessary. If you haven't tried increasing that, please do. You just need to add something like this to your project's manifest:

```json
	"creation": {
		"static": 65536
	}
```

**Reply by louisvangeldrop** [reply]
I still get the memory full error. Increasing the 64KB value with 1 KB steps has no positive effect. 

![image](https://user-images.githubusercontent.com/10716245/163691763-8201df51-7f41-40d3-83e1-1fe5f68a5e7d.png)

**Reply by phoddie** [reply]
There's quite a bit more memory in use than can be explained by normal TLS behavior. I'm afraid that you'll have to share a way to reproduce this problem for me to be able to offer further assistance with this.

**Reply by louisvangeldrop** [reply]
I have a Fujitsu Airco/heatpump with a builtin wifi-module and  I want to control it with the ioBroker application.
fglair-app is the official Fujitsu-app and it is a cloud based solution. I can access this cloud based solution using a nodejs-based solution (https://github.com/aanon4/homebridge-fujitsu-smart/blob/master/fglairAPI.js). This works now and I have integrated it within ioBroker using mqtt. However I did see it as a challenge to move this to Moddable. In the meantime however I also discovered that it is quite weird that I only can access the wifi-module in my network through polling a cloud server.
https://github.com/deiger/AirCon offers a local network solution. It is written in Python and lacks clear documentation. My next challenge is to find out how it works and see if I can port it to Moddable.

**Reply by phoddie** [reply]
It sounds like you may move to a different approach, one that sidesteps TLS. If you'd like to try to resolve this TLS issue, please provide a way to reproduce it. If not, that's fine – but please close out this ~issue~ discussion as there's no practical way to address it further. Thank you!

**Reply by louisvangeldrop** [reply]
Indeed I will go the local network approach. More appropriate and clean. So there is no need to investigate the memory problem.
